test/integration/shared-worker/shared-worker.test.ts

```diff
@@ -0,0 +1,740 @@
+
+      pubnubWithWorker.addListener({
+        status: (statusEvent) => {
+          if (statusEvent.category === PubNub.CATEGORIES.PNConnectedCategory && !messageReceived) {
+            // Wait a bit for subscription to be fully established before publishing
+            setTimeout(() => {
+              pubnubWithWorker
+                .publish({
+                  channel,
+                  message: testMessage,
+                })
+                .then((publishResult) => {
+                  expect(publishResult.timetoken).to.exist;
+                })
+                .catch(done);
+            }, 500);
+          }
+        },
+        message: (messageEvent) => {
+          if (!messageReceived && messageEvent.channel === channel) {
+            messageReceived = true;
+            try {
+              expect(messageEvent.channel).to.equal(channel);
+              expect(messageEvent.message).to.deep.equal(testMessage);
+              expect(messageEvent.timetoken).to.exist;
+              done();
+            } catch (error) {
+              done(error);
+            }
+          }
+        },
+      });
+
+      const subscription = pubnubWithWorker.channel(channel).subscription();
+      subscription.subscribe();
+    }).timeout(15000);
+
+    it('should handle subscription changes and receive messages on new channels', (done) => {
+      const channel1 = testChannels[0];
+      const channel2 = testChannels[1];
+      const testMessage = {
+        text: `Test message for channel 2: ${Date.now()}`,
+        sender: 'test-user',
+      };
+      let subscriptionReady = false;
+      let messageReceived = false;
+
+      pubnubWithWorker.addListener({
+        status: (statusEvent) => {
+          if (statusEvent.category === PubNub.CATEGORIES.PNConnectedCategory && !subscriptionReady) {
+            subscriptionReady = true;
+
+            // Wait for subscription to be fully established, then publish
+            setTimeout(() => {
+              pubnubWithWorker
+                .publish({
+                  channel: channel2,
+                  message: testMessage,
+                })
+                .catch(done);
+            }, 1000);
+          }
+        },
+        message: (messageEvent) => {
+          if (!messageReceived && messageEvent.channel === channel2) {
+            messageReceived = true;
+            try {
+              expect(messageEvent.channel).to.equal(channel2);
+              expect(messageEvent.message).to.deep.equal(testMessage);
+              done();
+            } catch (error) {
+              done(error);
+            }
+          }
+        },
+      });
+
+      // Subscribe to both channels at once since shared worker will aggregate them
+      const subscription1 = pubnubWithWorker.channel(channel1).subscription();
+      const subscription2 = pubnubWithWorker.channel(channel2).subscription();
+
+      subscription1.subscribe();
+      subscription2.subscribe();
```

Comment:
It will be great to have one channel subscribed to, the moment when we subscribe to another set. It can be two subscriptions subscribed and then before the third one, we unsubscribe from one of the subscriptions and immediately call subscribe on third one - for which message will be published and checked to receive.